NX generate fails due to incorrect `require` statement

### Current Behavior

 We have some code that calls `nx generate`; however, this code is failing with the error below. This appears to be caused by line [393 in `packages/nx/src/command-line/generate/generate.ts`](https://github.com/nrwl/nx/blob/127281b460073237f826bd4fedbbfa3947ac7804/packages/nx/src/command-line/generate/generate.ts#L393).
 
 That line is `require('../adapter/compat')`; however, that file is up a level and should be brought in via `require('../../adapter/compat'). This appears to have happened in #16460 when the files were reorganized.
 
 ```
NX  Generating ./dist/libs/components/i18n:lib-resources-module


 >  NX   Cannot find module '../adapter/compat'

 Error
    at ChildProcess.<anonymous> (C:\Blackbaud_Repos\skyux\node_modules\@skyux\dev-infra-private\lib\utils\spawn.js:52:24)
    at ChildProcess.emit (node:events:513:28)
    at cp.emit (C:\Blackbaud_Repos\skyux\node_modules\cross-spawn\lib\enoent.js:34:29)
    at ChildProcess._handle.onexit (node:internal/child_process:291:12)
```

### Expected Behavior

`nx generate` should not error.

### GitHub Repo

_No response_

### Steps to Reproduce

1. Run `nx generate` with a non-NX generator so that the if-statement at line 366 ( `if (ws.isNxGenerator(opts.collectionName, normalizedGeneratorName))` ) is not hit.


### Nx Report

```shell
Node   : 18.14.2
   OS     : win32 x64
   npm    : 9.5.0
   Hasher : Native
   
   nx                 : 16.2.1
   @nx/js             : 16.2.1
   @nx/jest           : 16.2.1
   @nx/linter         : 16.2.1
   @nx/workspace      : 16.2.1
   @nx/angular        : 16.2.1
   @nx/cypress        : 16.2.1
   @nx/devkit         : 16.2.1
   @nx/eslint-plugin  : 16.2.1
   @nx/node           : 16.2.1
   @nx/plugin         : 16.2.1
   @nx/storybook      : 16.2.1
   @nrwl/tao          : 16.2.1
   @nx/webpack        : 16.2.1
   nx-cloud           : 16.0.5
   typescript         : 4.9.5
   ---------------------------------------
   Community plugins:
   @storybook/angular : 7.0.9
   rxjs               : 6.6.7
   ---------------------------------------
   Local workspace plugins:
         @skyux-sdk/prettier-schematics
         @skyux/packages
         @skyux-sdk/e2e-schematics
```


### Failure Logs

_No response_

### Operating System

- [ ] macOS
- [ ] Linux
- [X] Windows
- [ ] Other (Please specify)

### Additional Information

_No response_

--- a/packages/nx/src/command-line/generate/generate.ts
+++ b/packages/nx/src/command-line/generate/generate.ts
@@ -390,7 +390,7 @@ export async function generate(cwd: string, args: { [k: string]: any }) {
         logger.warn(`\nNOTE: The "dryRun" flag means no changes were made.`);
       }
     } else {
-      require('../adapter/compat');
+      require('../../adapter/compat');
       return (await import('../../adapter/ngcli-adapter')).generate(
         workspaceRoot,
         {

